### ISSUE TITLE
Can entire Superagent API be used?

### ISSUE THREAD

**Issue by gsccheng** [visionmedia/supertest#431]
I tried a simple:
```
const request = require('supertest');

describe('/actions', () => {
  describe('GET /', () => {
    it('works', () => {
      request
        .get('/actions')
        .expect(200)
        .end(function(err, res) {
          if (err) throw err;
        });
    });
  });
})
```

and I get an error:
```
- TypeError: request.get is not a function
```
I want to use supertest without having the `app` object.

I'm using Supertest@3.0.0

**Comment by gsccheng** [comment]
Nevermind, you initialize `request` with the url.

**Comment by markstos** [comment]
The intent is that Superagent will largely work. Currently there is bug with the ".agent" behavior of superagent not working that is resolved by PR #453
